### DISCUSSION TITLE
Rerenders MUI DataGrid when updating the selection model

### DISCUSSION THREAD

**Discussion by adi-works** [pmndrs/zustand#1401] [Q&A]
I'm trying to use Zustand with MUI DataGrid, so I set up a store that has the table data state and the selection model state. When I update the selection model, which is just an array of row ids returned from the table, it's causing a re-render of the entire table. If I set up a local selection model state with `useState`, the issue is gone. Any ideas why?

```js
import create from "zustand";

export const useCustomersStore = create((set) => {
  return {
    customerList: [],
    selectedCustomersList: [],
    setCustomerList: (customerList) => {
      set({ customerList });
    },
    setSelectedCustomersList: (ids) => {
      set({ selectedCustomersList: ids });
    },
  };
});

```

**Comment by dai-shi** [comment]
Would be able to provide a reproduction as small as possible, preferably without using other libs?

**Reply by adi-works** [reply]
I can do that, yea. Most likely by tomorrow.

**Reply by ghost** [reply]
Don't procrastinate! you just end up forgetting :laugh:

**Reply by adi-works** [reply]
This ended up being a non-issue. ;)
